pkg/dcmtime/common.go

```diff
@@ -0,0 +1,147 @@
+package dcmtime
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// isIncluded returns whether `check` is included in `limit`.
+//
+// Example: to test whether seconds should be included, you would:
+// isIncluded(PrecisionSeconds, [caller-passed-limit])
+func isIncluded(check PrecisionLevel, precision PrecisionLevel) bool {
+	return check <= precision
+}
+
+// truncateMilliseconds truncate nanosecond time.Time value to arbitrary precision.
+func truncateMilliseconds(
+	nanoSeconds int, precision PrecisionLevel,
+) (millisecondsStr string) {
+	milliseconds := nanoSeconds / 1000
+	millisecondsStr = fmt.Sprintf("%06d", milliseconds)
+	millisecondsStr = millisecondsStr[:6-(PrecisionFull-precision)]
+
+	return millisecondsStr
+}
+
+// Const for time.Time timezone. We don't want a timezone because we don't REALLY know
+// that a TM or DA value is UTC, so we don't want UTC to be made explicit.
+var zeroTimezone = time.FixedZone("", 0)
+
+// durationInfo holds parsed piece of DA, TM, or DT info.
+type durationInfo struct {
+	// The parsed int value
+	Value int
+	// Whether it was present in the string
+	Present bool
+	// Only used when parsing Fractal seconds: the level of precision used.
+	FractalPrecision PrecisionLevel
+}
+
+// extractDurationInfo extracts a piece of DA, TM, or DT info from a parsed regex.
+func extractDurationInfo(subMatches []string, index int, fractal bool) (
+	info durationInfo, err error,
+) {
```

Comment:
Not so sure a linebreak is needed here. Also in this case, naming the return parameters doesn't provide any more information than the name of the types themselves, so I would remove them.

Reply:
In this case naming them lets me serves as a declaration so I can just set fields in the method, but I'll move the declaration into the body of the function.